lib/matplotlib/contour.py

```diff
@@ -400,8 +400,13 @@ def calc_label_rot_and_inline(self, slc, ind, lw, lc=None, spacing=5):
 
             ind = 0
 
-        # Path length in pixel space
-        pl = mlab.path_length(slc)
+        def _path_length(X):
+            dX = np.diff(X, axis=0)
+            # Alternative to np.linalg.norm()
+            # as numpy 1.7 doesn't have 'axis' kwarg
+            dX = np.sum(dX**2, axis=1)**(1 / 2)
+            return np.concatenate((np.zeros(1), np.cumsum(dX)))
```

Comment:
I suspect this can be done more efficiently with 3 lines in place of the function:
```
pl = np.zeros(slc.shape[0], dtype=float)
dx = np.diff(slc, axis=0)
pl[1:] = np.cumsum(np.hypot(dx[:, 0], dx[:, 1]))
```